### PR TITLE
GH-114 adding unit and integration tests

### PR DESCRIPTION
**Pull request by francoisledroff**
adding unit and integration tests see https://github.com/adobe/aio-lib-java/issues/114

### CODE REVIEW

**Review by sangeetha5491 — CHANGES_REQUESTED** [review]
First round of review

File: events_mgmt/src/test/java/com/adobe/aio/event/management/model/RegistrationInputModelTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        .description("some description")
+        .clientId("some client Id").build();
+  }
+
+  @Test
+  public void valid() {
+    String name = "com.adobe.aio.event.management.test.registration";
+    String description = name + " description";
+    String clientId = "some_clientId";
+    RegistrationInputModel registrationInputModel = RegistrationInputModel.builder()
```
[the remark this answers is not in the document]
Comment (francoisledroff) [reply events_mgmt/src/test/java/com/adobe/aio/event/management/model/RegistrationInputModelTest.java:43]: let's keep it simple for now, yes  Events API will returns the error anyway...
we could add it once we migrate to your V2 Registration API. wdyt ?

File: .github/workflows/maven-itest.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      # Set up environment with Java and Maven
+      - name: Setup JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: 11
+          cache: maven
+
+      # Build & Test
+      - name: Stage Integration Test with Maven
```
[the remark this answers is not in the document]
Comment (francoisledroff) [reply .github/workflows/maven-itest.yml]: done

File: core/README.md [hunk 3]
```diff
@@ -29,6 +29,17 @@ The `Workspace` POJO holds your Adobe Developer Console Project configurations
...
+### Private Key
+
+On top of these, the [`Workspace`](./src/main/java/com/adobe/aio/workspace/Workspace.java) POJO 
+can also hold your private Key 
+(associated with the public key you uploaded in your Adobe Developer Console Workspace)
+this will help to power Adobe JWT authentication flow and transparently add the proper `Bearer`
+authentication token to all your request, 
+Note the easiest way is to stuff your Private Key as a `pkcs8` base64 encoded String 
+using the `aio_encoded_pkcs8` property key.
+confer  [aio-lib-java-ims](../ims) documentation for more details.
```
[the remark this answers is not in the document]
Comment (francoisledroff) [reply core/README.md]: done

File: events_mgmt/src/main/java/com/adobe/aio/event/management/model/EventsOfInterest.java [hunk 4]
```diff
@@ -27,6 +28,9 @@ private EventsOfInterest() {
   }
 
   private EventsOfInterest(String providerId, String eventCode) {
+    if (StringUtils.isBlank(providerId) || StringUtils.isBlank(eventCode)){
+      throw new IllegalArgumentException("Invalid EventsOfInterest, nor providerId, nor eventCode can be empty/blank");
```
[the remark this answers is not in the document]
Comment (francoisledroff) [reply events_mgmt/src/main/java/com/adobe/aio/event/management/model/EventsOfInterest.java]: done

File: events_journal/src/test/java/com/adobe/aio/event/management/feign/FeignProviderServiceIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+        providerId);
+
+    Optional<Provider> providerById = providerService.findProviderById(providerId);
+    Assert.assertTrue(providerById.isPresent());
+    List<EventMetadata> eventMetadataList = providerService.getEventMetadata(providerId);
+    Assert.assertTrue(eventMetadataList.size() == 1);
+    Assert.assertEquals(eventMetadata.get(), eventMetadataList.get(0));
+    logger.info("Found AIO Events Provider `{}` by Id", providerById);
+
+    Optional<Provider> providerByInstanceId = providerService.findCustomEventsProviderByInstanceId(
```
Comment (sangeetha5491) [inline events_journal/src/test/java/com/adobe/aio/event/management/feign/FeignProviderServiceIntegrationTest.java:159]: for a custom events provider, is the instance id visible?

Comment (sangeetha5491) [reply events_journal/src/test/java/com/adobe/aio/event/management/feign/FeignProviderServiceIntegrationTest.java:159]: its something we set right?

why not just find by id for custom providers as well?

File: events_journal/src/test/java/com/adobe/aio/event/journal/feign/FeignJournalServiceIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+        pollingDuration += sleeptime;
+        if (pollingDuration < JOURNAL_POLLING_TIME_OUT_IN_MILLISECONDS) {
+          Thread.sleep(sleeptime);
+        } else {
+          logger.error("We polled the journal for " + JOURNAL_POLLING_TIME_OUT_IN_MILLISECONDS
+              + " milliseconds and could NOT find the expected eventId.");
+          return false;
+        }
+      }
+      entry = journalService.get(entry.getNextLink());
```
[the remark this answers is not in the document]
Comment (francoisledroff) [reply events_journal/src/test/java/com/adobe/aio/event/journal/feign/FeignJournalServiceIntegrationTest.java:144]: until `JOURNAL_POLLING_TIME_OUT_IN_MILLISECONDS`

File: events_journal/README.md [hunk 7]
```diff
@@ -9,26 +9,19 @@ by [`Adobe I/O Events` Journaling API](https://www.adobe.io/apis/experienceplatf
...
- You can use interceptors that our [ims](../ims) library provides,
- or any other [Open Feign RequestInterceptor](https://github.com/OpenFeign/feign#request-interceptors)
- adding :
-   * an `Authorization` header with a valid `Bearer` access token
-   * a `x-api-key` header associated with the above token
- * [2] set the workspace context expected by the ProviderService
- * [3] pass your registration journaling endpoint URL 
- * [4] have this service retrieve the oldest entry available in your registration journal 
- * [5] you can then loop to retrieve them all following entries' `next` links
+ * [1] build your ProviderService by passing your workspace context  (cf. the [aio-lib-java-core](../core) for workspace documentation)
```
[the remark this answers is not in the document]
Comment (francoisledroff) [reply events_journal/README.md:21]: done

File: events_journal/src/test/java/com/adobe/aio/event/management/feign/FeignProviderServiceIntegrationTest.java (hunk 5 above)
Comment (francoisledroff) [reply events_journal/src/test/java/com/adobe/aio/event/management/feign/FeignProviderServiceIntegrationTest.java:159]: yes we could have done it... but here I wanted to test 
* `Optional<Provider> findCustomEventsProviderByInstanceId(final String instanceId)` that calls
*  `Optional<Provider> findProviderBy(final String providerMetadataId, final String instanceId)`

**Review by sangeetha5491 — APPROVED** [review]

**Pull request merged**